### PR TITLE
Darrin eide/map reference to paper

### PR DESCRIPTION
**Pull request by DarrinEide**
See readme.md file for full details on the tool.

Sample output is included in sampleData/sample-output.txt

### CODE REVIEW

File: src/PAK-Samples/MapReferenceToPaper/Program.cs [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+        {
+            var resultString = GetQueryInterpretations(
+                query: referenceString,
+                subscriptionKey: subscriptionKey,
+                complete: false,
+                interpretationCount: 1,
+                entityCount: 1,
+                attributes: attributesToMap + "," + attributesToReturn,
+                
+                // NOTE: Longer timeouts provide better quality interpretations. 500 is typically
```
Comment (RickRogahn) [inline src/PAK-Samples/MapReferenceToPaper/Program.cs]: >500 is typically [](start = 81, length = 16)

is typically...? #Resolved

File: src/PAK-Samples/MapReferenceToPaper/Program.cs [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+
+            return null;
+        }
+
+        /// <summary>
+        /// Generates interpretation of query using Project Academic Knowledge Interpret method
+        /// </summary>
+        /// <param name="query">Natural language query to interpret</param>
+        /// <returns>JSON string of the Interpret method response. See https://docs.microsoft.com/en-us/academic-services/project-academic-knowledge/reference-interpret-method</returns>
+        static string GetQueryInterpretations(
```
Comment (RickRogahn) [inline src/PAK-Samples/MapReferenceToPaper/Program.cs:271]: >GetQueryInterpretations [](start = 22, length = 23)

Wouldn't these requests get throttled eventually?  #Closed

File: src/PAK-Samples/MapReferenceToPaper/Program.cs [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+            string query,
+            string subscriptionKey,
+            bool complete = false,
+            int interpretationCount = 5,
+            int entityCount = 0,
+            string attributes = "",
+            int timeout = 500
+            )
+        {
+            var client = new HttpClient();
```
Comment (RickRogahn) [inline src/PAK-Samples/MapReferenceToPaper/Program.cs]: >var client = new HttpClient(); [](start = 12, length = 30)

Can this be created once and shared?  If I remember correctly each instance of HttpClient holds onto a socket connection even after disposed for ~2mins.  .NET Core solved this with an HTTPClientFactory but I don't think it is available in .NET Framework.  #Resolved

File: src/PAK-Samples/MapReferenceToPaper/Program.cs (hunk 2 above)
Comment (DarrinEide) [reply src/PAK-Samples/MapReferenceToPaper/Program.cs:271]: Yes

---
In reply to: [395925547](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395925547) [](ancestors = 395925547)

Comment (RickRogahn) [reply src/PAK-Samples/MapReferenceToPaper/Program.cs:271]: Should we self throttle then?

---
In reply to: [395927328](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395927328) [](ancestors = 395927328,395925547)

File: src/PAK-Samples/MapReferenceToPaper/readme.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# Map Academic Reference Strings to Microsoft Academic paper entities
+
+This folder contains sample code for generating academic reference strings from tabular input data and mapping the reference strings to [Microsoft Academic paper entities](https://docs.microsoft.com/en-us/academic-services/project-academic-knowledge/reference-paper-entity-attributes).
+
+## Prerequisites
+
+* Free subscription for the [Project Academic Knowledge](https://msr-apis.portal.azure-api.net/products/project-academic-knowledge) Academic Search API
+* Microsoft Visual Studio C#
```
Comment (RickRogahn) [inline src/PAK-Samples/MapReferenceToPaper/readme.md]: >Studio [](start = 19, length = 6)

C# compiler (Visual Studio or Visual Studio Code recommended) #Resolved

File: src/PAK-Samples/MapReferenceToPaper/readme.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+## Building the project
+
+1. Navigate to the MapReferenceToPaper directory
+1. Open the MapReferenceToPaper.sln file
+1. Once Visual Studio opens, build the project by going to Build->Build Solution
```
Comment (RickRogahn) [inline src/PAK-Samples/MapReferenceToPaper/readme.md]: >by going to Build->Build Solution [](start = 47, length = 33)

Suggestion: select Build->Build Solution from the top menu. #Resolved

Comment (RickRogahn) [inline src/PAK-Samples/MapReferenceToPaper/readme.md]: >by going to Build->Build Solution [](start = 47, length = 33)

Suggestion:  Would it be better for them to compile the project and use the .exe that is generated as the example and document the parameters to pass? #Resolved

File: src/PAK-Samples/MapReferenceToPaper/Program.cs (hunk 2 above)
Comment (DarrinEide) [reply src/PAK-Samples/MapReferenceToPaper/Program.cs:271]: They will eventually get throttled by the 10k/month limit, but stopping them before that would be the same as them hitting the limit unless I'm missing your point..

---
In reply to: [395927503](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395927503) [](ancestors = 395927503,395927328,395925547)

Comment (RickRogahn) [reply src/PAK-Samples/MapReferenceToPaper/Program.cs:271]: I guess the question is, can someone run this script w/o hitting the 10K limit?  If not, we should probably point out that out and maybe direct them to MAKES.

---
In reply to: [395932827](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395932827) [](ancestors = 395932827,395927503,395927328,395925547)

Comment (DarrinEide) [reply src/PAK-Samples/MapReferenceToPaper/Program.cs:271]: The sample data has ~2k rows, so yes

---
In reply to: [395935469](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395935469) [](ancestors = 395935469,395932827,395927503,395927328,395925547)

Comment (DarrinEide) [reply src/PAK-Samples/MapReferenceToPaper/Program.cs:271]: Also, the current version of MAKES does not support the new default grammar yet, so we unfortunately can't point them to it just yet

---
In reply to: [395935469](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395935469) [](ancestors = 395935469,395932827,395927503,395927328,395925547)

**Review by RickRogahn — APPROVED** [review]
:shipit:

**Pull request merged**